activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -141,6 +141,8 @@ def read_entry(key, **options)
 
         # Write an entry to the cache.
         def write_entry(key, entry, **options)
+          # Prevents duplicate compression within Dalli.
+          options = options.merge(compress: false)
```

Comment:
Passing `compress: false` directly as a keyword argument is a bit more efficient:

```patch
diff --git a/activesupport/lib/active_support/cache/mem_cache_store.rb b/activesupport/lib/active_support/cache/mem_cache_store.rb
index f646c6ff82..ae3102406d 100644
--- a/activesupport/lib/active_support/cache/mem_cache_store.rb
+++ b/activesupport/lib/active_support/cache/mem_cache_store.rb
@@ -149,7 +149,7 @@ def write_entry(key, entry, **options)
             expires_in += 5.minutes
           end
           rescue_error_with false do
-            @data.with { |c| c.send(method, key, value, expires_in, **options) }
+            @data.with { |c| c.send(method, key, value, expires_in, **options, compress: false) }
           end
         end
```

```bash
$ ruby -r benchmark-memory -e "def foo(**) end; options = {}; Benchmark.memory { |x| x.report { options = options.merge(compress: false); foo(**options) } }"
Calculating -------------------------------------
                       672.000  memsize (   168.000  retained)
                         4.000  objects (     1.000  retained)
                         0.000  strings (     0.000  retained)
$ ruby -r benchmark-memory -e "def foo(**); end; options = {}; Benchmark.memory { |x| x.report { foo(**options, compress: false) } }"
Calculating -------------------------------------
                       336.000  memsize (     0.000  retained)
                         2.000  objects (     0.000  retained)
                         0.000  strings (     0.000  retained)
```